File: IronVault/IronVault.Model/Models/KorisnikTrener.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace IronVault.Model.Models
{
    public partial class KorisnikTrener
    {
        public int KorisnikTrenerId { get; set; }

        public int KorisnikId { get; set; }

        public int TrenerId { get; set; }

        public DateTime DatumTermina { get; set; }

        public int ZakazanoSati { get; set; }

        public virtual Korisnik Korisnik { get; set; } = null!;

        public virtual Trener Trener { get; set; } = null!;
    }
}

File: IronVault/UI/ironvault_mobile/lib/models/korisnik_trener.dart
import 'package:ironvault_mobile/models/korisnik.dart';
import 'package:ironvault_mobile/models/trener.dart';
import 'package:json_annotation/json_annotation.dart';

part 'korisnik_trener.g.dart';

@JsonSerializable()
class KorisnikTrener {
  int? korisnikTrenerId;
  int? korisnikId;
  Korisnik? korisnik;
  int? trenerId;
  Trener? trener;
  DateTime? datumTermina;
  int? zakazanoSati;

  KorisnikTrener({this.korisnikTrenerId});

  factory KorisnikTrener.fromJson(Map<String, dynamic> json) =>
      _$KorisnikTrenerFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$KorisnikTrenerToJson(this);
}
